File: Low Level Design (LLD)/Problems/02. ParkingLot/Java/factory/VehicleFactory.java
package factory;

import model.*;

public class VehicleFactory {
    public static Vehicle createVehicle(VehicleType type, String license) {
        switch (type) {
            case CAR : return new Car(license);
            case BIKE : return new Bike(license);
            case TRUCK : return new Truck(license);
            default: return null;
        }
    }
}

File: Low Level Design (LLD)/Problems/02. ParkingLot/Cpp/factory/src/VehicleFactory.cpp
#include "../include/VehicleFactory.h"

Vehicle* VehicleFactory::createVehicle(VehicleType type, const std::string& license) {
    switch (type) {
        case VehicleType::CAR:
            return new Car(license); 
        case VehicleType::BIKE:
            return new Bike(license);
        case VehicleType::TRUCK:
            return new Truck(license);
        default:
            return nullptr; 
    }
}
